src/s2wasm.h

```diff
@@ -395,20 +403,55 @@ class S2WasmBuilder {
       skipWhitespace();
       if (debug) dump("process");
       if (!*s) break;
-      if (*s != '.') break;
+      if (*s != '.') parseAssignment();
+      else {
+        s++;
+        if (match("text")) parseText();
+        else if (match("type")) parseType();
+        else if (match("weak") || match("hidden") || match("protected") || match("internal")) getStr(); // contents are in the content that follows
+        else if (match("imports")) skipImports();
+        else if (match("data")) {}
+        else if (match("ident")) skipToEOL();
+        else if (match("section")) parseToplevelSection();
+        else if (match("align") || match("p2align")) skipToEOL();
+        else if (match("globl")) parseGlobl();
+        else if (match("functype")) parseFuncType();
+        else abort_on("process");
+      }
+    }
+  }
+
+  void parseAssignment() {
+    if (debug) dump("assignment");
+    Name lhs = getStrToSep();
+    skipEqual();
+
+    // get the relocation
+    Name rhs = getStrToSep();
+    assert(!isFunctionName(rhs));
+    size_t offset = 0;
+    if (*s == '+') {
       s++;
-      if (match("text")) parseText();
-      else if (match("type")) parseType();
-      else if (match("weak") || match("hidden") || match("protected") || match("internal")) getStr(); // contents are in the content that follows
-      else if (match("imports")) skipImports();
-      else if (match("data")) {}
-      else if (match("ident")) skipToEOL();
-      else if (match("section")) parseToplevelSection();
-      else if (match("align") || match("p2align")) skipToEOL();
-      else if (match("globl")) parseGlobl();
-      else if (match("functype")) parseFuncType();
-      else abort_on("process");
+      offset = getInt();
     }
+    skipWhitespace();
+
+    // get the data size
+    mustMatch(".size");
+    mustMatch(lhs.str);
+    mustMatch(",");
+    size_t size = atoi(getStr().str);
+    skipWhitespace();
+
+    // copy over the data
+    const std::vector<char> *src = linkerObj->findRelocationData(rhs);
+    if (!src) Fatal() << "Unknown relocation assignment: " << rhs << '\n';
+    assert(offset + size <= src->size());
+    std::vector<char> data(src->begin() + offset, src->begin() + offset + size);
```

Comment:
Yes, a symbol assignment just assigns a value to a symbol, and does not create any actual program data of its own.